service/service.go

```diff
@@ -427,15 +435,23 @@ func (s *Service) Info() (*InfoResponse, error) {
 	return res, nil
 }
 
-func (s *Service) newRound(epoch uint32) *round {
+func (s *Service) newRound(epoch uint32) {
 	if err := s.saveState(); err != nil {
 		panic(err)
 	}
-	r := newRound(s.sig, s.cfg, s.datadir, epoch)
+	r := newRound(s.sig, s.datadir, epoch)
 	if err := r.open(); err != nil {
 		panic(fmt.Errorf("failed to open round: %v", err))
 	}
-	return r
+
+	s.openRound = r
+}
+
+func (s *Service) openRoundID() string {
+	s.openRoundMutex.Lock()
+	defer s.openRoundMutex.Unlock()
+
+	return s.openRound.ID
```

Comment:
This method might be called concurrently from many incoming GRPC requests and it just reads. Please consider using an `RWMutex` to allow many concurrent readers. (If you don't go with the atomic.Pointer)